### PR TITLE
Measurement UI 

### PR DESCRIPTION
**Pull request by jagankumar-egov**

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/ContractCardDetails.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        "Work Order Number:": contractNumber,
+        "Project Id:": projectID,
+        "Project Sanction Date:": "", // did not find the value yet
+        "Project Name:": projectName,
+        "Project Description:": projectDesc,
+        "Project Location:": projectLoc,
+    };
+
+    return (
+        <div className="contract-details-card" style={cardStyle}>
```
[the remark this answers is not in the document]
Comment (jagankumar-egov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/ContractCardDetails.js:56]: add in measurement scss

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureCard.js [hunk 2]
```diff
@@ -1,20 +1,144 @@
...
+        return { ...state, tableState: [...state.tableState, newRow] };
+      case "UPDATE_ROW":
+        const {
+          state: { id, value, row, type },
+        } = action;
+        const { tableState } = state;
+        let findIndex = tableState.findIndex((row, index) => {
+          return index + 1 === id;
+        });
+        if (type === "number") tableState[findIndex].number = value;
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureCard.js:34]: revisit this multiple if conditions

@rachana-egov

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureCard.js [hunk 3]
```diff
@@ -1,20 +1,144 @@
...
+                  tableData[tableKey][tableIndex].measures = state.tableState;
+                  setValue("measurements", tableData);
+
+                  setInitialState(state);
+                  setConsumedQty(total);
+                  setShowMeasureCard(false);
+                }} />
+              </div>
+            </td>
+            <td colSpan={"4"}>
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureCard.js:134]: localisation code
@himanshukeshari-eGov  @prerna-egov

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureRow.js [hunk 4]
```diff
@@ -1,20 +1,66 @@
...
   const history = useHistory();
 
 
   return (
-    <Card>
-        View
-    </Card>
+    <tr key={index}>
+      <td>{state.tableState[index].sNo}</td>
+      <td>{state.tableState[index].isDeduction ? "YES" : "NO"}</td>
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureRow.js:15]: localisation code
@himanshukeshari-eGov  @prerna-egov

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureTable.js [hunk 5]
```diff
@@ -1,20 +1,191 @@
...
+          <tr>{renderHeader()}</tr>
+        </thead>
+        <tbody>{renderBody()}
+
+        </tbody>
+
+      </table>
+      <div style={{ display: "flex", flexDirection: "row", justifyContent: "flex-end", margin: "20px" }}>
+        <div style={{ display: "flex", flexDirection: "row", fontSize: "1.2rem" }}>
+          <span style={{ fontWeight: "bold" }}>Total MB Amount(For Current Entry): </span>
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureTable.js:182]: localisation code
@himanshukeshari-eGov  @prerna-egov

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasurementCard.js [hunk 6]
```diff
@@ -1,32 +1,56 @@
...
+      {
+        label: t("MB_CREATE"),
+        link: `/${window?.contextPath}/employee/measurement/create`,
+      },
+      {
+        label: t("MB_INBOX"),
+        link: `/${window?.contextPath}/employee/measurement/inbox`,
+      },
       {
         label: t("ACTION_TEST_CREATE_MEASUREMENT"),
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasurementCard.js:39]: update the first create link and remove this entry
@rachana-egov

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/configs/MeasurementInboxConfig.js [hunk 7]
```diff
@@ -1,190 +1,233 @@
...
+    type: "inbox",
+    // actionLabel: "WORKS_ADD_WAGESEEKER",
+    // actionRole: "INDIVIDUAL_CREATOR",
+    // actionLink: "masters/create-wageseeker",
+    apiDetails: {
+      serviceName: "/inbox/v2/_search",
+      requestParam: {},
+      requestBody: {
+        inbox: {
+          tenantId: Digit.ULBService.getCurrentTenantId(),
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/configs/MeasurementInboxConfig.js:13]: add at ui customization and move this config to mdms
@himanshukeshari-eGov

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ResponseBanner.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    // Now you can work with the response object as needed
+
+    // const isApplicationSubmitted = responseObj?.get?.responseInfo?.status === "successful";
+
+
+    const goToHome = () => {
+
+
+        history.push({
+            pathname: "/digit-ui/employee",
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ResponseBanner.js:49]: update the path names

Comment (jagankumar-egov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ResponseBanner.js:49]: it should have context path as dynamic

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ResponseBanner.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+                    <div className="link " >
+                        <ArrowRightInbox style={{ width: '24px', height: '24px' }} />
+                        <Link to={`/${window.contextPath}/employee/measurement/inbox`}>
+                            {"MB_GO_INBOX"}
+                        </Link>
+                    </div>
+
+                    <div className="link">
+                        <AddNewIcon style={{ width: '24px', height: '24px' }} />
+                        <Link to={`/${window.contextPath}/employee/measurement/inbox`}>
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ResponseBanner.js:85]: add new reading should go to create screen again

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/UpdateMeasurement.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+                console.error('Error:', error);
+            }
+        };
+
+        // Trigger the data fetching when the component mounts
+        fetchData();
+    }, []);
+
+    return (
+        <div>
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/UpdateMeasurement.js:37]: add the logic to integrate with Create screen to edit and update@rachana-egov

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ViewMeasurement.js [hunk 11]
```diff
@@ -1,20 +1,99 @@
...
+
+  const projectDetails = {"applicationDetails" : [applicationDetails?.applicationDetails?.applicationDetails[0]]};
+  const imageDetails = {"applicationDetails" : [applicationDetails?.applicationDetails?.applicationDetails[2]]};
+
+
+  const sorEstimates = applicationDetails?.applicationData?.estimate[0]?.estimateDetails.filter(item => item.category === 'SOR');
+  const nonSorEstimates = applicationDetails?.applicationData?.estimate[0]?.estimateDetails.filter(item => item.category === 'NONSOR');
+
+  const measures = applicationDetails?.applicationData?.measurements[0].measures
+  console.log(measures, "measure")
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ViewMeasurement.js:37]: remove logs @prerna-egov

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ViewMeasurement.js [hunk 12]
```diff
@@ -1,20 +1,99 @@
...
+    }
+  }
+
+  let { isLoading, isError, data: applicationDetails, error } = Digit.Hooks.measurement.useViewMeasurement(tenantId, criteria);
+
+  const projectDetails = {"applicationDetails" : [applicationDetails?.applicationDetails?.applicationDetails[0]]};
+  const imageDetails = {"applicationDetails" : [applicationDetails?.applicationDetails?.applicationDetails[2]]};
+
+
+  const sorEstimates = applicationDetails?.applicationData?.estimate[0]?.estimateDetails.filter(item => item.category === 'SOR');
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ViewMeasurement.js:33]: update the logics

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ViewMeasurement.js [hunk 13]
```diff
@@ -1,20 +1,99 @@
...
+        workflowDetails={applicationDetails?.workflowDetails}
+        mutate={() => {}}
+        tenantId={tenantId}
+        showTimeLine={false}
+      />
+      <MeasurementHistory contractNumber={workOrderNumber} />
+      <Card className="override-card view-utilization-card">
+        <Button 
+        className={"view-utilization-btn"} 
+        label={"View Utilisation Statement"}>
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ViewMeasurement.js:72]: localisation code
@himanshukeshari-eGov  @prerna-egov

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/utils/transformData.js [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+                    sorAmount: data.sumSor || 0,
+                    nonSorAmount: data.sumNonSor || 0,
+                    totalAmount: (data.sumSor ? data.sumSor : 0) + (data.sumNonSor ? data.sumNonSor : 0),
+                },
+                "wfStatus": "DRAFTED",
+                "workflow": {
+                    "action": "SAVE_AS_DRAFT",
+                    "assignes": [],
+                    "comments": "string",
+                    "verificationDocuments": [
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/utils/transformData.js:94]: remove

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/utils/transformMeasurementData.js [hunk 15]
```diff
@@ -0,0 +1,45 @@
+
+
+export const transformMeasurementData = async (mbNumber) => {
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/utils/transformMeasurementData.js:3]: add comments to each functions @prerna-egov @rachana-egov

**Review by jagankumar-egov — COMMENTED** [review]
check all the comments and integrate with the BFF service 
@rachana-egov @prerna-egov @himanshukeshari-eGov @ansh-egov

File: frontend/micro-ui/web/micro-ui-internals/example/src/UICustomizations.js [hunk 16]
```diff
@@ -611,15 +611,239 @@ export const UICustomizations = {
...
+    "sortBy": "string",
+    "order": "DESC"
+      };
+      data.body.params = {};
+    
+      return data;
+      
+    },
+    additionalCustomizations: (row, key, column, value, t, searchResult) => {
+      console.log(key,value);
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/example/src/UICustomizations.js:642]: remove logs

File: frontend/micro-ui/web/micro-ui-internals/example/src/UICustomizations.js [hunk 17]
```diff
@@ -611,15 +611,239 @@ export const UICustomizations = {
...
+    preProcess: (data) => {
+      //set tenantId
+      data.body.inbox.tenantId = Digit.ULBService.getCurrentTenantId();
+      data.body.inbox.processSearchCriteria.tenantId = Digit.ULBService.getCurrentTenantId();
+      const estimateNumber = data?.body?.inbox?.moduleSearchCriteria?.estimateNumber?.trim()
+      if(estimateNumber) data.body.inbox.moduleSearchCriteria.estimateNumber = estimateNumber
+      const projectId = data?.body?.inbox?.moduleSearchCriteria?.projectId?.trim()
+      if(projectId) data.body.inbox.moduleSearchCriteria.projectId = projectId
+      // deleting them for now(assignee-> need clarity from pintu,ward-> static for now,not implemented BE side)
+      const assignee = _.clone(data.body.inbox.moduleSearchCriteria.assignee);
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/example/src/UICustomizations.js:703]: update these logics 
@himanshukeshari-eGov

File: frontend/micro-ui/web/micro-ui-internals/example/src/UICustomizations.js (hunk 16 above)
Comment (himanshukeshari-eGov) [reply frontend/micro-ui/web/micro-ui-internals/example/src/UICustomizations.js:642]: Done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MBHistoryTable.js [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+            </tr>
+          ))}
+        </tbody>
+      </table>
+
+      </CustomCollapsibleTable>
+      <Button 
+      className={"collapse-button"}
+      onButtonClick={toggleTableCollapse}
+      label={isTableCollapsed ? 'Show MB History' : 'Hide MB History'}>
```
[the remark this answers is not in the document]
Comment (himanshukeshari-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MBHistoryTable.js:139]: Done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureCard.js [hunk 19]
```diff
@@ -1,20 +1,144 @@
...
+      <table className="table reports-table sub-work-table" >
+        <thead>
+          <tr>{renderHeader()}</tr>
+        </thead>
+        <tbody>
+          {renderBody()}
+          <tr>
+            <td colSpan={"4"}>
+              <div style={{ display: "flex", flexDirection: "row" }}>
+                <Button label={"Clear"} onButtonClick={() => {
```
[the remark this answers is not in the document]
Comment (himanshukeshari-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureCard.js:121]: Done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureCard.js (hunk 3 above)
Comment (himanshukeshari-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureCard.js:134]: Done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureRow.js (hunk 4 above)
Comment (himanshukeshari-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureRow.js:15]: Done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ViewMeasurement.js (hunk 13 above)
Comment (himanshukeshari-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ViewMeasurement.js]: Done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ResponseBanner.js (hunk 9 above)
Comment (himanshukeshari-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/pages/employee/ResponseBanner.js:85]: Done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureTable.js (hunk 5 above)
Comment (himanshukeshari-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/Measurement/src/components/MeasureTable.js:181]: Done

**Pull request closed without merging**